### ISSUE TITLE
Irish insular features for mga (Middle Irish) and ghc (Hiberno-Scottish Gaelic)

### ISSUE THREAD

**Issue by silmeth** [psb1558/Junicode-font#170]
When using Junicode Two Beta with `ss02` for a HTML element with `lang="ga"` (Irish) or `lang="gd"` (Scottish Gaelic) or `lang="sga"` (Old Irish), the text uses the “Irish” features (eg. the Irish form of Insular ꝺ).

But it isn’t so if the language of the text is set to `mga` (Middle Irish, the form of Goidelic dialects used between ~900–1200) or `ghc` (“Hiberno-Scottish Gaelic”, a weird name assigned to Classical Gaelic, literary/poetic standard of the Gaelic language of ~1200–1750, by Ethnologue¹).

Check the codes `mga` and `ghc` in [BCP47 Language Subtag Lookup](https://r12a.github.io/app-subtags/).

I’d expect `mga` and `ghc` to be displayed the same way as Old Irish and Irish.

----
¹ [Ethnologue 15, p. 565](https://archive.org/details/ethnologuelangua0000unse_g7e1/page/564/mode/):
> **Gaelic, Hiberno-Scottish** (Gaoidhealg, Hiberno-Scottish Classical Common Gaelic) [ghc] Extinct. Ireland and Scotland. *Class:* Indo-European, Celtic, Insular, Goidelic. *Lg Dev:* Roman script. Bible: 1690. *Other:* Archaic literary language based on 12th century Irish, formerly used by professional classes in Ireland until the 17th century and Scotland until the 18th century. vso.

**Comment by psb1558** [comment]
Unfortunately, these language tags are not part of the [OpenType font specification](https://learn.microsoft.com/en-us/typography/opentype/spec/languagetags), which provides only a subset of the language tags in the relevant standards. Junicode Two has all the available tags for Irish and Gaelic: IRI, IRT, SGA, GAE.

There's a link to a form for suggesting additional language tags on the page linked above.

**Comment by silmeth** [comment]
Ah, I was afraid that might be some OpenType limitation. Thanks for the response!

**Comment by silmeth** [comment]
Just a self-note that the CSS attribute [`font-language-override`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-language-override) with the appropriate language system tag (`"IRT"` for Irish traditional, `"SGA"` for Old Irish) is a valid work-around for this:

```html
<span lang="ghc" style="font-language-override:&quot;IRT&quot;">A Ṡionann Ḃríain Ḃóroiṁe,<br/>
iongnaḋ is méd do ġáire,<br/>
mar sguire dod ġlóraiġe<br/>
ag dol síar isin sáile.</span>
```

![A Shionann Bhríain Bhóroimhe…](https://user-images.githubusercontent.com/5329644/233783651-1680fa01-50c2-4c59-a4d5-0c1e01001878.png)

**Comment by psb1558** [comment]
Very elegant (FireFox is often the first to implemen new CSS features). BTW, should the dot be lowered on the insular cap B?

**Comment by silmeth** [comment]
Yes, I think in most old Gaelic types the capital B is basically scaled-up lowercase b (and thus the dot placement is similar to the lowercase variant). [This](https://wikisource.org/wiki/Page:T%C3%A1in_B%C3%B3_Cuailnge_%27na_dr%C3%A1ma_-_Ua_Laoghaire.pdf/5) for example is either capitals or small caps (as it uses the capital “Roman” R rather than the insular r seen below in *Forbaḋ*):
![MUINTIR NA LEAḂAR GAEḊILGE](https://user-images.githubusercontent.com/5329644/233784283-5032f27a-2491-4316-989d-9ec00ef65ab5.png)

or [here](https://wikisource.org/wiki/Page:T%C3%A1in_B%C3%B3_Cuailnge_%27na_dr%C3%A1ma_-_Ua_Laoghaire.pdf/16):
![longer passage from *Táin ’na dhráma*](https://user-images.githubusercontent.com/5329644/233784617-1be1b985-fc50-442c-b063-c9db9c9fd543.png)

You can see examples of older metal Irish types on [Gaelchló website](https://www.gaelchlo.com/miotal.html) (in Irish) – each paragraph ends in a link to a PDF with example of a page printed with that type. But in Early Modern Irish you wouldn’t have a sentence beginning with a ⟨ḃ⟩ (in Modern Irish *Ḃí* ‘was’ is common at the start, but in Classical Gaelic/EMI it’s rather *Do-ḃí* or *Do-ḃáoi*, or similar), so I don’t think there is any example there.

**Comment by psb1558** [comment]
Okay, thanks: I've moved the dot.

**Comment by psb1558** [comment]
I'll leave this open and add those languages when/if they make it into the OpenType standard.
